package_manager/contracts/PR/zeppelin/NewZosPackage.sol

```diff
@@ -0,0 +1,66 @@
+pragma solidity ^0.4.24;
+
+import '../RepoPackage.sol';
+import 'openzeppelin-solidity/contracts/ownership/Ownable.sol';
+import 'zos-lib/contracts/application/versioning/ImplementationProvider.sol';
+import 'zos-lib/contracts/application/versioning/ImplementationDirectory.sol';
+
+/**
+ * @title NewZosPackage
+ * @dev Collection of contracts grouped into versions.
+ * Contracts with the same name can have different implementation addresses in different versions.
+ */
+contract NewZosPackage is Ownable, RepoPackage {
+  /**
+   * @dev Emitted when a version is added to the package.
+   * XXX The version is not indexed due to truffle testing constraints.
+   * @param version Name of the added version.
+   * @param provider ImplementationProvider associated with the version.
+   */
+  event VersionAdded(uint16[3] version, ImplementationProvider indexed provider);
```

Comment:
Why is it the responsibility of this contract to emit an event with the `provider`? Should the base `RepoPackage` be emitting this info along with the URI?

Reply:
Agree, this should not be part of ZosPackage.

Reply:
Sure, we should move that to `RepoPackage`. Didn't want to do that in the PoC just to make sure there was an implementation satisfying both specs :)